internal/hcl/project_locator.go

```diff
@@ -232,18 +232,27 @@ func (p *ProjectLocator) walkPaths(fullPath string, level int) []string {
 	}
 
 	files := hclParser.Files()
-	var providerBlocks bool
+	var hasProviderBlock bool
+	var hasBackendAttr bool
 
 	for _, file := range files {
-		body, content, diags := file.Body.PartialContent(justProviderBlocks)
+		body, content, diags := file.Body.PartialContent(terraformAndProviderBlocks)
 		if diags != nil && diags.HasErrors() {
 			p.logger.Warn().Err(diags).Msgf("skipping building module information for file %s as failed to get partial body contents", file)
 			continue
 		}
 
-		// only do this after looping through all files
-		if len(body.Blocks) > 0 {
-			providerBlocks = true
+		providerBlocks := body.Blocks.OfType("provider")
+		if len(providerBlocks) > 0 {
+			hasProviderBlock = true
+		}
+
+		terraformBlocks := body.Blocks.OfType("terraform")
+		for _, block := range terraformBlocks {
+			backend, _, _ := block.Body.PartialContent(nestedBackendBlock)
+			if len(backend.Blocks) > 0 {
+				hasBackendAttr = true
+			}
```

Comment:
I believe this will work:

```suggestion
			if block.Type == "backend" {
				hasBackendAttr = true
				break
			}
```

And this means we don't need to decode the nested block, which is a somewhat heavy process

Reply:
scratch that - this won't work as we need the nested body.